### PR TITLE
Fix SCons interface

### PR DESCRIPTION
**Pull request by pablogs9**
Signed-off-by: Pablo Garrido <pablogs9@gmail.com>

### CODE REVIEW

File: extra_script.py [hunk 1]
```diff
@@ -83,36 +83,39 @@
...
-    global_env['LIBS'].append(builder.library_name)
+    global_env.Append(LIBS=[builder.library_name])
 
 # Add library path
-global_env['LIBPATH'].append(builder.library_path)
+global_env.Append(LIBPATH=[builder.library_path])
 
 # Add required defines
-global_env['_CPPDEFFLAGS'] += ' -DCLOCK_MONOTONIC=0 '
+global_env.Append(CCFLAGS=['-DCLOCK_MONOTONIC=0'])
```
Comment (ivankravets) [inline extra_script.py:92]: Please use `global_env.Append(CPPDEFINES=[("CLOCK_MONOTONIC", 1)])`. 

A list of `(macro, value)` tuples.

File: extra_script.py [hunk 2]
```diff
@@ -83,36 +83,39 @@
...
+env.Append(CPPPATH=[
+    main_path + "/platform_code",
+    main_path + "/platform_code/{}/{}".format(framework, microros_transport)])
 
 # Add micro-ROS defines to user application
-global_env['_CPPDEFFLAGS'] += ' -DMICRO_ROS_TRANSPORT_{}_{} '.format(framework.upper(), microros_transport.upper())
-global_env['_CPPDEFFLAGS'] += ' -DMICRO_ROS_DISTRO_ '.format(microros_distro.upper())
+global_env.Append(CCFLAGS=[
+    '-DMICRO_ROS_TRANSPORT_{}_{}'.format(framework.upper(), microros_transport.upper()),
+    '-DMICRO_ROS_DISTRO_ {} '.format(microros_distro.upper())])
```
Comment (ivankravets) [inline extra_script.py:113]: `CPPDEFINES` scope, no need to specify `-D`.

**Example**
```python
env.Append(CPPDEFINES=[("CLOCK_MONOTONIC", 1), MICRO_ROS_TRANSPORT_{}_{}'.format(framework.upper(), microros_transport.upper())]
```

**Review by ivankravets — CHANGES_REQUESTED** [review]

File: extra_script.py (hunk 1 above)
Comment (pablogs9) [reply extra_script.py]: Addressed here: 8c2818bf6bce86a4b57b38b01db72642ec57cbd0

File: extra_script.py (hunk 2 above)
Comment (pablogs9) [reply extra_script.py]: Addressed here: 8c2818bf6bce86a4b57b38b01db72642ec57cbd0

File: extra_script.py (hunk 1 above)
Comment (pablogs9) [reply extra_script.py]: Addressed here: 4b894e194f1a8e29379928f0f90c2f652e50a17a

File: extra_script.py (hunk 2 above)
Comment (pablogs9) [reply extra_script.py]: Addressed here: 4b894e194f1a8e29379928f0f90c2f652e50a17a

**Pull request merged**